pilot/cmd/pilot-discovery/main.go

```diff
@@ -66,6 +67,12 @@ var (
 			// Create the stop channel for all of the servers.
 			stop := make(chan struct{})
 
+			// If keepaliveMaxServerConnectionAge is negative, istiod crash
+			// https://github.com/istio/istio/issues/27257
+			if err := validation.ValidateMaxServerConnectionAge(serverArgs.KeepaliveOptions.MaxServerConnectionAge); err != nil {
```

Comment:
I think you should call this in a PreRun or similar cobra section, not RunE. Not a cobra expert, @esnible probably knows more

Reply:
Agree, will do that.

Reply:
Done.